ui/Screen/Project.svelte

```diff
@@ -45,10 +45,7 @@
   };
 
   onMount(() => {
-    localPeer.projectEvents.subscribe(event => {
-      if (!event) {
-        return;
-      }
```

Comment:
Just for my personal understanding, this new dependency, and the extra infra around it, is pulled in to remove these three lines? The point has been made before that the approach of manually subscribing to stores in components is subpar in general. This screen is already backed by a store which can be equipped to do the subscription to meaningful events and only update the store accordingly.

While the Svelte store might look like an abuse of the mechanism it was modeled with the single original use-case of powering the connectivity indicator. If there is a need for streams than it would look more natural to use the `EventSource` which natively offers dispatch semantics. It can't be exposed raw as it needs parsing and such, but that gives an opportunity to rethink the public API of `localPeer`, which could easily expose a `subscribe` function allowing to pass a list of events that should be emitted. This truly would bring utility to the rest of the codebase imho.

Reply:
> this new dependency, and the extra infra around it, is pulled in to remove these three lines?

It is also pulled in to remove these three lines but not only. The “null” pattern can be eliminated everywhere where events are handled. In addition we remove the “latest event” issue where any subscription to the events immediately calls the subscriber with the last event—even if that event was triggered a long time ago.

> The point has been made before that the approach of manually subscribing to stores in components is subpar in general. This screen is already backed by a store which can be equipped to do the subscription to meaningful events and only update the store accordingly.

As I mentioned in my reply to @cloudhead, this PR is not to fix where we subscribe to events but how to provide better facilities to manage events. I agree with you and @cloudhead that the subscription should happen closer to the store.

> If there is a need for streams than it would look more natural to use the `EventSource` which natively offers dispatch semantics.

Could you elaborate on what you mean by dispatch semantics? I see a couple of problems with `EventSource` that make in unergonomic to use. 1. It does not support combinators that allow for say easy filtering. 2. The `addEventListener`/`removeEventListener` API creates a mismatch with sveltes `subscribe() => unsubscribe` pattern. 3. The EventSource is recreated throughout the life time of the app which makes using it directly cumbersome.

> gives an opportunity to rethink the public API of `localPeer`, which could easily expose a `subscribe` function allowing to pass a list of events that should be emitted.

This is definitely worth exploring. But I could be harder to create properly typed APIs for such a generic `subscribe` function.